python/revng/scripting/projects/cli_project.py

```diff
@@ -0,0 +1,97 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import shutil
+import subprocess
+from tempfile import NamedTemporaryFile, TemporaryDirectory
+from typing import Any, List
+
+import yaml
+
+from revng.model import Binary, dump_diff, load_model_from_string, make_diff
+
+from .project import Project
+
+
+class CLIProject(Project):
+    def __init__(self, working_dir: str = "", revng_executable: str = ""):
+        self.revng_executable = revng_executable
+        self.working_dir = working_dir
+        self.input_binary = ""
+        if not self.working_dir:
+            self._tmp_dir = TemporaryDirectory()
+            self.working_dir = self._tmp_dir.name
+        if not self.revng_executable:
+            self.revng_executable = "revng"
+            assert shutil.which(self.revng_executable) is not None
+        super().__init__()
+
+    def import_binary(self, input_binary_path: str) -> None:
+        self.input_binary = input_binary_path
+
+    def get_artifact(self, artifact_name: str) -> bytes | str:
+        return self._get_artifact_internal(artifact_name)
+
+    def _get_artifact_internal(self, artifact_name: str, targets_class: List[Any] = []) -> bytes | str:
```

Comment:
`Any` should be used only in exceptional situations, here for instance is better to use `TypeVar("T", bound=StructBase)`. Also check all the other instances of `Any`.